src/ray/raylet/node_manager.cc

```diff
@@ -3371,19 +3366,25 @@ void NodeManager::HandleGetNodeStats(const rpc::GetNodeStatsRequest &request,
   // HandleGetNodeStats should set a timeout so that the rpc finishes even if not all
   // workers have replied.
   auto all_workers = worker_pool_.GetAllWorkers();
+  absl::flat_hash_set<WorkerID> driver_ids;
+  for (auto driver : worker_pool_.GetAllDrivers()) {
+    all_workers.push_back(driver);
+    driver_ids.insert(driver->WorkerId());
```

Comment:
Tests might be failing since we are now requiring successful RPCs to all drivers. Potential workaround is to drop failed RPCs without hanging the return.